internal/vsp/feepayment.go

```diff
@@ -325,10 +351,8 @@ func (fp *feePayment) next() time.Duration {
 	w := fp.client.wallet
 	_, tipHeight := w.MainChainTip(fp.ctx)
 
-	fp.mu.Lock()
-	ticketLive := fp.ticketLive
-	ticketExpires := fp.ticketExpires
-	fp.mu.Unlock()
+	ticketLive := fp.liveHeight()
+	ticketExpires := fp.expiryHeight()
```

Comment:
we probably want to ensure that both of these are calculated together with the mutex being held, instead of releasing it after liveHeight() and acquiring it a second time.